File: ServicioRestProveedorVS/WSProveedorSMSApi/WSProveedorSMSApi/Models/Proveedor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WSProveedorSMSApi.Models
{
    public class Proveedor
    {      
        public int _id { get; set; }
        public String _nombre { get; set; }
        public String _numero { get; set; }
        public bool _disponible { get; set; }

        public Proveedor(int id, String nombre, String numero, bool disponible)
        {
            _id = id;
            _nombre = nombre;
            _numero = numero;
            _disponible = disponible;
        }
    }
}

File: appSideMenu/src/models/Proveedor.ts
/** Modelo para la clase Proveedor */
export class Proveedor {

    private id: number;
    private nombre: string;
    private numero: string;
    private disponible: boolean;

    constructor(data?: any) {
        if (data !== undefined) {
            this.$id = data._id;
            this.$nombre = data._nombre;
            this.$numero = data._numero;
            this.$disponible = data._disponible;
        }
    }

    /** Getters y setters */
    public get $id(): number {
        return this.id;
    }

    public set $id(value: number) {
        this.id = value;
    }

    public get $nombre(): string {
        return this.nombre;
    }

    public set $nombre(value: string) {
        this.nombre = value;
    }

    public get $numero(): string {
        return this.numero;
    }

    public set $numero(value: string) {
        this.numero = value;
    }

    public get $disponible(): boolean {
        return this.disponible;
    }

    public set $disponible(value: boolean) {
        this.disponible = value;
    }
}
